src/main/java/io/gravitee/common/spring/factory/SpringFactoriesLoader.java

```diff
@@ -53,49 +48,18 @@ public void setApplicationContext(ApplicationContext applicationContext) throws
     protected Collection<? extends T> getFactoriesInstances() {
         if (factories == null) {
             logger.debug("Loading instances for type {}", getObjectType().getName());
-            factories = getSpringFactoriesInstances(getObjectType(), new Class<?>[] {});
+            factories = createSpringFactoriesInstances(applicationContext, getObjectType());
         } else {
             logger.debug("Instances for type {} already loaded. Skipping...", getObjectType().getName());
         }
 
         return factories;
     }
 
-    private Collection<? extends T> getSpringFactoriesInstances(Class<T> type, Class<?>[] parameterTypes, Object... args) {
-        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
```

Comment:
There is a specific mechanism for loading classes and beans for plugins. By removing this part you may break the plugin spring beans loading, inheritance, and separations. Are you able to confirm/infirm this in some way ?

Reply:
This is why I try multiples deployments. I haven't a global view on how it's used in gravitee. I don't know make more tests.

I make this part only to cleanup the file (don't use deprecated method), but I can reduce the changeset to the minimum to be more confident

Reply:
I have splited in 2 commits one with minimal fix and another with improvments